File: technical_skills_java/src/main/java/com/natiqhaciyef/technical_skills_java/leetcode/interview/RemoveElement.java
package com.natiqhaciyef.technical_skills_java.leetcode.interview;

public class RemoveElement {

    public static int removeElement(int[] nums, int val) {
        int count = 0;

        for (int current = nums.length - 1; current > 0; current--) {
            if (nums[current] != val) {
                for (int next = current - 1; next >= 0; next--) {
                    if (nums[next] == val) {
                        nums[next] = nums[current];
                        nums[current] = val;
                        System.out.println("Next: " + nums[next]);
                        System.out.println("Current: " + nums[current]);
                    }
                }
            }
        }

        for (int num : nums) {
            if (num != val)
                count += 1;
        }

        return count;
    }

    public static void main(String[] args) {
        int[] arr = {3, 2, 2, 3};
        int[] arr2 = {0, 1, 2, 2, 3, 0, 4, 2};
        System.out.println(removeElement(arr2, 2));
    }
}

File: technical_skills_kotlin/src/main/java/com/natiqhaciyef/technical_skills_kotlin/leetcode/interview/Remove Element.kt
package com.natiqhaciyef.technical_skills_kotlin.leetcode.interview


fun removeElement(nums: IntArray, element: Int): Int {
    for (current in nums.size - 1 downTo 1) {
        if (nums[current] != element) {
            for (next in current - 1 downTo 0) {
                if (nums[next] == element) {
                    nums[next] = nums[current]
                    nums[current] = element
                }
            }
        }
    }

//    print("[")
//    var isFirst = true
//
//    for (index in nums.indices) {
////        if (nums[index] != element) {
//        if (index == 0 || isFirst) {
//            print("${nums[index]}")
//            isFirst = false
//        } else {
//            print(",${nums[index]}")
////            }
//        }
//    }
//    print("]")
//    println()

    return nums.count { it != element }
}


fun main() {
    val nums = intArrayOf(3, 2, 2, 3)
    val nums2 = intArrayOf(0, 1, 2, 2, 3, 0, 4, 2)
    println(removeElement(nums2, 2))
}